scipy/linalg/tests/test_sketches.py

```diff
@@ -0,0 +1,72 @@
+"""Tests for _sketches.py."""
+
+from __future__ import division, print_function, absolute_import
+import numpy as np
+from scipy.linalg import clarkson_woodruff_transform
+
+__usage__ = """
+Build linalg:
+  python setup_linalg.py build
+Run tests if scipy is installed:
+  python -c 'import scipy;scipy.linalg.test()'
+"""
+
+from numpy.testing import (assert_equal, assert_almost_equal,
+                           assert_array_almost_equal, assert_array_equal,
+                           assert_raises, assert_, assert_allclose)
+
+import pytest
+
+
+def make_random_dense_gaussian_matrix(n_rows, n_columns, mu=0, sigma=0.01):
+    """
+    Make some random data with Gaussian distributed values
+    """
+    res = np.random.normal(mu, sigma, n_rows*n_columns)
+    return np.reshape(res, (n_rows, n_columns))
+
+class TestClarksonWoodruffTransform(object):
+    """
+    Testing the Clarkson Woodruff Transform
+    """
+    # Big dense matrix dimensions
+    n_matrix_rows = 2000
+    n_matrix_columns = 100
+
+    # Sketch matrix dimensions
+    n_sketch_rows = 100
+
+    # Repetitions/max_errors per test
+    repetitions_per_test = 10
+    n_max_errors = 3
+
+    # Error threshold
+    threshold = 0.1
+
+    dense_big_matrix = make_random_dense_gaussian_matrix(n_matrix_rows,
+        n_matrix_columns)
+    
+    def test_sketch_dimensions(self):
+        sketch = clarkson_woodruff_transform(
+            self.dense_big_matrix,
+            self.n_sketch_rows)
+
+        assert(sketch.shape == (
```

Comment:
Please use ``assert_`` from ``numpy.testing`` instead of ``assert`` (plain ``assert``s are not allowed in scipy, because they get stripped with ``python -OO``.

Reply:
I did not know that. Changed.